lib/pbench/cli/server/tree_manage.py

```diff
@@ -0,0 +1,72 @@
+from typing import List
+
+import click
+
+from pbench.cli.server import config_setup, pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server import BadConfig
+from pbench.server.filetree import FileTree
+
+
+def print_tree(tree: FileTree):
+    print(f"Tree anchored at {tree.archive_root}\n")
+
+    if len(tree.datasets) == 0 and len(tree.controllers) == 0:
+        print("Pbench file tree is empty")
+        return
+
+    print("Tarballs:")
+    for tarball in tree.datasets.values():
+        print(f"  {tarball.name}")
+
+    print("\nControllers:")
+    for controller in tree.controllers.values():
+        print(f"  Controller {controller.name}:")
+        for tarball in controller.tarballs.values():
+            print(f"    Tarball {tarball.name}")
+            if tarball.unpacked:
+                print(f"      Unpacked in {tarball.unpacked}")
+                states: List[str] = []
+                for name, path in controller.state_dirs.items():
+                    for link in path.iterdir():
+                        if tarball.name in link.name:
+                            states.append(name)
+                if states:
+                    states.sort()
+                    print(f"        States: {', '.join(states)}")
+
+
+@click.command(name="pbench-tree-manager")
+@pass_cli_context
+@click.option(
+    "--display", default=False, is_flag=True, help="Display the full tree on completion"
+)
+@common_options
+def tree_manage(context: object, display: bool):
+    """
+    Discover, display, and manipulate the on-disk representation of controllers
```

Comment:
You're right; there were originally options to create, unpack, and "uncache" (my best attempt at a name for removing the unpacked directory tree); which I removed either on my own because they weren't that useful or after discussion with Webb because the syntax was a bit weird.

I'll fix this if I do another commit; on the other hand, after this goes in I want to remove `pbench-server-prep-shim-002` entirely, which will require some other mechanism to install datasets from the intake directory in legacy `unittest` tarballs, and that probably means putting a better `create` mechanism here, so the code will catch up (again) to the stale comment. 😆